### PR TITLE
Add SteamCMD

### PR DESCRIPTION
**Pull request by gasinvein**
I'm not sure if SteamCMD should be provided as extension, but as most users don't need it, it's better to make it optional.
Todo:
- [x] Reuse steam-wrapper.py with steamcmd
- [x] Make it so that steamcmd desktop entry appears only if extension is installed
- [x] Add appdata for SteamCMD
- [x] Add different icon for SteamCMD

### CODE REVIEW

**Comment by nanonyme** [comment]
Something to consider with steam-wrapper:
1) Throw the setup.py stuff away, use regular executable
2) Make it instead of always pointing to steam have first argument be the command
3) Make entrypoint be a shell-script that will call steam-wrapper with steam as first argument

**Comment by gasinvein** [comment]
I thought of importing some functions from steam-wrapper directly. Is it bad idea?

**Comment by nanonyme** [comment]
I dunno. My thought experiment was basically making it use the first argument it gets as parameter. This would allow using it like 
#!/app/bin/steam-helper bash
exec steam $*

**Comment by nanonyme** [comment]
That would basically make it so that migration would be run before following script and all environment variables would be set

**Comment by nanonyme** [comment]
Maybe. I've just been leaning towards @TingPing's opinion that maybe entangling setuptools and pip into all of this may have been overkill. The main reason why there's a special copytree and a lot of other things (instead of some bash as originally) is the Cloud Sync fix was quite complex to get right and required specifically set timestamps in specific places. I dunno. I think it feels strange to have main functions calling each other. I think at this point it might be cleaner if the bootstrap and execution logic was split into a separate helper that would always take a path and the entrypoint of steam-wrapper would call that with path to steam

**Comment by nanonyme** [comment]
(plus yeah, I expected Cloud Sync would go wrong and there might need to be remediation migrations to fix things up which would have been painful in bash)

**Comment by gasinvein** [comment]
I agree with this, but it's feels out of scope here. Maybe we should create another branch with steam_wrapper.py rework?

**Comment by nanonyme** [comment]
Why would steamcmd have a desktop icon, btw? I thought it was a command line tool

**Comment by gasinvein** [comment]
Easy way to launch it. It has it's own command-line prompt, so I find useful ability to launch it in a terminal window from desktop menu.

**Comment by nanonyme** [comment]
Unless I'm misunderstanding something badly, isn't what you created a module, not extension? (https://blog.tingping.se/2018/03/18/flatpaking-plugins.html)

**Comment by gasinvein** [comment]
It's both. SteamCMD is built as a module, then exported separately from main app as an extension.

**Comment by gasinvein** [comment]
@nanonyme Did you mean someting like [this](https://github.com/flathub/com.valvesoftware.Steam/compare/master...gasinvein:wrapper-rework) should be done with steam-wrapper?

**Comment by nanonyme** [comment]
Yeah, something like that. It was originally just another file in steam module, it could go back there. Except instead of separate steam_wrapper.sh could as well use script source

**Comment by nanonyme** [comment]
Anyway. Strange. Looks like when autodownload is disabled, the module is fully installed except without the source tarball downloading. Hence the desktop icon is installed too

**Comment by gasinvein** [comment]
>instead of separate steam_wrapper.sh could as well use script source

Can yo explain it, please? My english not very good.

**Comment by nanonyme** [comment]
{
  "type": "script",
  "dest-filename": "steam-wrapper.sh",
   "commands": ["#!/app/bin/steam-helper bash", "exec $*"]
}
Or somesuch

**Comment by nanonyme** [comment]
Afaik the syntax is designed for doing trivial snippets without needing separate files

**Comment by gasinvein** [comment]
Is it possible? I thought flatpak-builder always appends `#!/bin/sh` to script type sources.

**Comment by nanonyme** [comment]
Started doubting that as you said. Anyway, I'm still not quite sure you can do that conditional install of desktop file

File: steam_wrapper/steamcmd_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    steamcmd_path = os.path.join(
+        steamcmd_install_dir, "steamcmd.sh"
+    )
+
+    if not os.path.isfile(steamcmd_path):
+        if os.path.isfile(os.path.join(STEAMCMD_BOOTSTRAP_DIR, "steamcmd.sh")):
+            steam_wrapper.copytree(
+                STEAMCMD_BOOTSTRAP_DIR,
+                steamcmd_install_dir,
```
Comment (nanonyme) [inline steam_wrapper/steamcmd_wrapper.py]: Doesn't this effectively mean uninstalling the app will not remove steamcmd?

Comment (gasinvein) [reply steam_wrapper/steamcmd_wrapper.py]: Yes. But I don't see any way to avoid this.

Comment (nanonyme) [reply steam_wrapper/steamcmd_wrapper.py]: I see what you mean.

**Comment by nanonyme** [comment]
If you can't get the conditional download and conditional desktop to work, I suggest you either download always or remove the entire desktop file. The logic you want would most likely easily work with a real app extension

**Comment by nanonyme** [comment]
I think we can as well just drop the extension thing if it's providing too much trouble. It's not as if steamcmd is that massive anyway. It's like just 6MB

**Comment by gasinvein** [comment]
What if we symlink `/app/share/applications/steamcmd.desktop` to `/app/steamcmd/share/applications/steamcmd.desktop`? Thus without extension installed the desktop file will be a broken symlink. Will it break anything, e.g. desktop menu?

**Comment by nanonyme** [comment]
I don't think /app/steamcmd/share/applications/steamcmd.desktop has anything to do with it, it's the desktop file that you install in build-commands that creates the visible desktop entry

**Comment by gasinvein** [comment]
I missed that, what's wrong with conditional download?

**Comment by nanonyme** [comment]
Well, with the conditional download get quickly exiting prompt with failure if the cmd is not installed if you try to open the start menu entry. It would imo be more consistent that either the start menu entry does something useful or doesn't exist at all

**Comment by nanonyme** [comment]
I'd say extension. Let's just keep it simple

**Comment by gasinvein** [comment]
Done. Now we need new appdata I guess.

**Comment by nanonyme** [comment]
Is there some generic free to use terminal icon that could be used? Anything Steam-related is most likely heavily trademarked so we should only use pristine images

**Comment by gasinvein** [comment]
Does flatpak require icon to be packaged in bundle? If not, we can use e.g. generic `terminal` icon or so.

**Comment by nanonyme** [comment]
I would assume it is not allowed as it imposes a dependency to host

**Comment by gasinvein** [comment]
I doubt there is a single linux desktop without a terminal icon.

**Comment by gasinvein** [comment]
`utilities-terminal` icon is present in all default icon themes, including HighContrast. I thing we can rely on it.

**Comment by nanonyme** [comment]
Sounds fine to me

**Comment by nanonyme** [comment]
bot, build

**Comment by AsciiWolf** [comment]
Sorry, but I really dislike that SteamCMD is installed by default as a part of the Steam Flatpak. It definitely should be an addon or a separate Flatpak.

**Comment by nanonyme** [comment]
Addons don't really work apparently, base apps and inheritance might do it

**Comment by gasinvein** [comment]
@AsciiWolf do you dislike desktop entry or the very presence of steamcmd?

**Comment by AsciiWolf** [comment]
@gasinvein The desktop entry.

**Comment by nanonyme** [comment]
I'm fine either way for desktop entry. You can make a PR if you like. Steamcmd itself has been requested for inclusion before this. Note that I cannot currently test any PR's due to broken machine

**Comment by AsciiWolf** [comment]
[Here](209) is a PR to remove the desktop file.

**Pull request merged**